NanoAnalysis/python/nanoZZ4lAnalysis.py

```diff
@@ -309,9 +315,10 @@
                           'keep HTXS_Higgs*',
                           'keep HTXS_njets30',
                           'keep Pileup*',
+                          'keep GenJet_*',
+                          'keep LHEMela*', 
```

Comment:
I suppose LHEMela and LHEPart are needed only when ADD_ALLEVENTS == True? If so can you please remove this line and L321. Otherwise, you should remove them from L331 and L334.